src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.InProcess.cs

```diff
@@ -133,50 +88,38 @@ internal abstract partial class AbstractNavigateToSearchService
+
             // Namespaces are never returned in nav-to as they're too common and have too many locations.
             if (declaredSymbolInfo.Kind == DeclaredSymbolInfoKind.Namespace)
                 continue;
 
-            await AddResultIfMatchAsync(
-                documentKey, document,
-                declaredSymbolInfo,
-                nameMatcher, containerMatcher,
-                kinds, onResultFound, cancellationToken).ConfigureAwait(false);
-        }
-    }
+            using var nameMatches = TemporaryArray<PatternMatch>.Empty;
+            using var containerMatches = TemporaryArray<PatternMatch>.Empty;
 
-    private static async Task AddResultIfMatchAsync(
-        DocumentKey documentKey,
-        Document? document,
-        DeclaredSymbolInfo declaredSymbolInfo,
-        PatternMatcher nameMatcher,
-        PatternMatcher? containerMatcher,
-        DeclaredSymbolInfoKindSet kinds,
-        Func<RoslynNavigateToItem, Task> onResultFound,
-        CancellationToken cancellationToken)
-    {
-        using var nameMatches = TemporaryArray<PatternMatch>.Empty;
-        using var containerMatches = TemporaryArray<PatternMatch>.Empty;
-
-        cancellationToken.ThrowIfCancellationRequested();
-        if (kinds.Contains(declaredSymbolInfo.Kind) &&
-            nameMatcher.AddMatches(declaredSymbolInfo.Name, ref nameMatches.AsRef()) &&
-            containerMatcher?.AddMatches(declaredSymbolInfo.FullyQualifiedContainerName, ref containerMatches.AsRef()) != false)
-        {
-            // See if we have a match in a linked file.  If so, see if we have the same match in
-            // other projects that this file is linked in.  If so, include the full set of projects
-            // the match is in so we can display that well in the UI.
-            //
-            // We can only do this in the case where the solution is loaded and thus we can examine
-            // the relationship between this document and the other documents linked to it.  In the
-            // case where the solution isn't fully loaded and we're just reading in cached data, we
-            // don't know what other files we're linked to and can't merge results in this fashion.
-            var additionalMatchingProjects = await GetAdditionalProjectsWithMatchAsync(
-                document, declaredSymbolInfo, cancellationToken).ConfigureAwait(false);
-
-            var result = ConvertResult(
-                documentKey, document, declaredSymbolInfo, nameMatches, containerMatches, additionalMatchingProjects);
-            await onResultFound(result).ConfigureAwait(false);
+            if (kinds.Contains(declaredSymbolInfo.Kind) &&
+                nameMatcher.AddMatches(declaredSymbolInfo.Name, ref nameMatches.AsRef()) &&
+                containerMatcher?.AddMatches(declaredSymbolInfo.FullyQualifiedContainerName, ref containerMatches.AsRef()) != false)
+            {
+                if (cancellationToken.IsCancellationRequested)
```

Comment:
>if (cancellationToken.IsCancellationRequested) [](http://example.com/codeflow?start=16&length=46)

nit: seems a bit overkill as this was checked earlier inside the loop

Reply:
i kept this as addmatches is the expensive step.  so this lets us bail after doing that check.